projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionResultManager.java

```diff
@@ -123,7 +124,8 @@ private static ExecutionTerminationStatus updateTerminationStatusBasedOnExecutio
       if (StringUtils.isEmpty(terminationStatusString) && ExecutionStatus.FINISHED.equals(executionStatus)) {
          terminationStatus = ExecutionTerminationStatus.SUCCESS;
       } else if (StringUtils.isEmpty(terminationStatusString) && ExecutionStatus.FAILED.equals(executionStatus)) {
-         if (StringUtils.equalsIgnoreCase(terminationReason, TERMINATION_REASON_DEADLINE_EXCEEDED)) {
+         if (StringUtils.equalsIgnoreCase(terminationReason, TERMINATION_REASON_DEADLINE_EXCEEDED) ||
+                 StringUtils.equalsIgnoreCase(terminationReason, TERMINATION_REASON_OUT_OF_MEMORY)) {
```

Comment:
Now it is user error. In what way would we indicate to the end-user , if it was deadline exceeded or memory. 

In other words we need to follow https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/CONTRIBUTING.md#error-handling

Reply:
In my opinion we should split the FAILED status into USER_ERROR and PLATFORM_ERROR. Then we will be able to fill in the _message_ field correctly - e.g. "OOM".

Reply:
Created issue for this here: https://github.com/vmware/versatile-data-kit/issues/484